### PR TITLE
Add missing profile location Facebook

### PR DESCRIPTION
**Pull request by jetaimefrc**

### CODE REVIEW

**Comment by YasharF** [comment]
Can you provide a reference to the part of the FB dev documentation about this?

File: config/passport.js [hunk 1]
```diff
@@ -138,6 +138,7 @@ passport.use(new FacebookStrategy({
           user.profile.name = user.profile.name || `${profile.name.givenName} ${profile.name.familyName}`;
           user.profile.gender = user.profile.gender || profile._json.gender;
           user.profile.picture = user.profile.picture || `https://graph.facebook.com/${profile.id}/picture?type=large`;
+          user.profile.location = (profile._json.location) ? profile._json.location.name : '';
```
Comment (YasharF) [inline config/passport.js]: Missing a || condition:  If the user has set their location in, user.profile.location, the default behavior needs to not overwrite the user's location.

Comment (jetaimefrc) [reply config/passport.js]: > Can you provide a reference to the part of the FB dev documentation about this?

Sorry, I recently received the notification. It [here](https://developers.facebook.com/docs/graph-api/reference/user/)

> Missing a || condition: If the user has set their location in, user.profile.location, the default behavior needs to not overwrite the user's location.
Okie, I fixed

**Comment by YasharF** [comment]
I tried this patch and it caused the following error

```
diff --git a/config/passport.js b/config/passport.js
index 03702766..fd2045b3 100644
--- a/config/passport.js
+++ b/config/passport.js
@@ -138,6 +138,7 @@ passport.use(new FacebookStrategy({
           user.profile.name = user.profile.name || `${profile.name.givenName} ${profile.name.familyName}`;
           user.profile.gender = user.profile.gender || profile._json.gender;
           user.profile.picture = user.profile.picture || `https://graph.facebook.com/${profile.id}/picture?type=large`;
+          user.profile.location = user.profile.location || profile._json.location.name;
           user.save((err) => {
             req.flash('info', { msg: 'Facebook account has been linked.' });
             done(err, user);
```


```
2019-07-28T03:08:01.810298+00:00 app[web.1]: GET /auth/facebook 302 9.602 ms - 0
2019-07-28T03:08:02.380532+00:00 app[web.1]: events.js:180
2019-07-28T03:08:02.380551+00:00 app[web.1]: throw er; // Unhandled 'error' event
2019-07-28T03:08:02.380553+00:00 app[web.1]: ^
2019-07-28T03:08:02.380554+00:00 app[web.1]:
2019-07-28T03:08:02.380556+00:00 app[web.1]: TypeError: Cannot read property 'name' of undefined
2019-07-28T03:08:02.380558+00:00 app[web.1]: at /app/config/passport.js:141:83
2019-07-28T03:08:02.380559+00:00 app[web.1]: at /app/node_modules/mongoose/lib/model.js:4567:16
2019-07-28T03:08:02.380561+00:00 app[web.1]: at /app/node_modules/mongoose/lib/query.js:4314:12
2019-07-28T03:08:02.380562+00:00 app[web.1]: at /app/node_modules/mongoose/lib/query.js:2797:28
2019-07-28T03:08:02.380564+00:00 app[web.1]: at processTicksAndRejections (internal/process/task_queues.js:75:11)
2019-07-28T03:08:02.380566+00:00 app[web.1]: Emitted 'error' event at:
2019-07-28T03:08:02.380568+00:00 app[web.1]: at /app/node_modules/mongoose/lib/model.js:4569:13
2019-07-28T03:08:02.380569+00:00 app[web.1]: at /app/node_modules/mongoose/lib/query.js:4314:12
2019-07-28T03:08:02.380571+00:00 app[web.1]: at /app/node_modules/mongoose/lib/query.js:2797:28
2019-07-28T03:08:02.380572+00:00 app[web.1]: at processTicksAndRejections (internal/process/task_queues.js:75:11)
2019-07-28T03:08:02.397082+00:00 app[web.1]: npm ERR! code ELIFECYCLE
2019-07-28T03:08:02.397282+00:00 app[web.1]: npm ERR! errno 1
2019-07-28T03:08:02.398182+00:00 app[web.1]: npm ERR! hackathon-starter@5.1.4 start: `node app.js`
2019-07-28T03:08:02.398279+00:00 app[web.1]: npm ERR! Exit status 1
2019-07-28T03:08:02.398409+00:00 app[web.1]: npm ERR!
2019-07-28T03:08:02.398500+00:00 app[web.1]: npm ERR! Failed at the hackathon-starter@5.1.4 start script.
2019-07-28T03:08:02.398594+00:00 app[web.1]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
2019-07-28T03:08:02.405841+00:00 app[web.1]:
2019-07-28T03:08:02.406045+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
2019-07-28T03:08:02.406133+00:00 app[web.1]: npm ERR!     /app/.npm/_logs/2019-07-28T03_08_02_399Z-debug.log
```

**Comment by YasharF** [comment]
In my attempt, the response from FB, did not have a .location.  I do have a location set for my FB profile, so can you please find the dev documentation that states the conditions which cause the location field to be returned or not returned?

**Comment by jetaimefrc** [comment]
> In my attempt, the response from FB, did not have a .location. I do have a location set for my FB profile, so can you please find the dev documentation that states the conditions which cause the location field to be returned or not returned?

You can check this to see!
https://graph.facebook.com/v4.0/me?fields=id,name,birthday,email,gender,first_name,last_name,location&access_token=EAAEx6LpE2VkBAC3PCvjqnZCLHrhGzewoqc9LY4HH539ZA0G2tZBYZCgQ1MSMyuga0OzFXS3UzqJld0AseUfBdjWDaeuZAVVo7aChsQoKNbHLkbh3u3giZCjL19xT44V8AtYdHzOz0wN7aW7x7NQbNcBK6NyWDZCXDN1io7NT9v32AZDZD

Sorry about that, I use it:

```js
user.profile.location = (profile.location) ? profile.location.name : '';
```

**DOCUMENT**
https://developers.facebook.com/docs/graph-api/reference/location/

I tested in my laptop! It's worked!

**Comment by YasharF** [comment]
> In my attempt, the response from FB, did not have a .location. I do have a location set for my FB profile, so can you please find the dev documentation that states the conditions which cause the location field to be returned or not returned?

> I tried this patch and it caused an error

**Pull request closed without merging**